inference/models/paligemma/paligemma.py

```diff
@@ -150,6 +161,36 @@ def download_model_artefacts_from_s3(self) -> None:
         raise NotImplementedError()
 
 
+class LoRAPaliGemma(PaliGemma):
+    def __init__(self, model_id, *args, huggingface_token=HUGGINGFACE_TOKEN, **kwargs):
+        self.huggingface_token = huggingface_token
+        super().__init__(model_id, *args, **kwargs)
+
+    def initialize_model(self):
+        lora_config = LoraConfig.from_pretrained(self.cache_dir, device_map=DEVICE)
+        model_id = lora_config.base_model_name_or_path
+        revision = lora_config.revision
+        self.dtype = torch.float16
+        if revision is not None:
+            self.dtype = getattr(torch, revision)
+        base_cache_dir = os.path.join(MODEL_CACHE_DIR, "huggingface")
+        if self.huggingface_token is None:
+            raise RuntimeError(
+                "Must set environment variable HUGGINGFACE_TOKEN to load LoRA "
+                "(or pass huggingface_token to this __init__)"
+            )
+        self.base_model = PaliGemmaForConditionalGeneration.from_pretrained(
+            model_id,
+            revision=revision,
+            device_map=DEVICE,
+            cache_dir=base_cache_dir,
+            token=self.huggingface_token,
+        ).to(self.dtype)
+        self.model = get_peft_model(self.base_model, lora_config).eval().to(self.dtype)
+
+        self.processor = AutoProcessor.from_pretrained(self.cache_dir)
+
+
 if __name__ == "__main__":
```

Comment:
I know that not part of the change - but is that needed?

Reply:
sorry, is what needed? the __main__? no, it was just for testing, we can remove

Reply:
yeah, main